worker-macros/src/event.rs

```diff
@@ -124,6 +128,40 @@ pub fn expand_macro(attr: TokenStream, item: TokenStream) -> TokenStream {
 
             TokenStream::from(output)
         }
+        #[cfg(feature = "queue")]
+        Queue => {
+            // save original fn name for re-use in the wrapper fn
+            let input_fn_ident = Ident::new(
+                &(input_fn.sig.ident.to_string() + "_queue_glue"),
+                input_fn.sig.ident.span(),
+            );
+            let wrapper_fn_ident = Ident::new("queue", input_fn.sig.ident.span());
+            // rename the original attributed fn
+            input_fn.sig.ident = input_fn_ident.clone();
+
+            let wrapper_fn = quote! {
+                pub async fn #wrapper_fn_ident(event: ::worker::worker_sys::MessageBatch, env: ::worker::Env, ctx: ::worker::worker_sys::Context) {
+                    // call the original fn
+                    let ctx = worker::Context::new(ctx);
+                    #input_fn_ident(::worker::MessageBatch::from(event), env, ctx).await
```

Comment:
Like `fetch` handler I think we should have the user's `queue` handler return a `Result<()>`.

Reply:
Done, I've updated it to return `Result<()>`